File: server/src/main/java/net/tachyon/network/netty/codec/PacketEncoder.java
package net.tachyon.network.netty.codec;

import io.netty.buffer.ByteBuf;
import io.netty.channel.ChannelHandlerContext;
import io.netty.handler.codec.MessageToByteEncoder;
import net.tachyon.network.packet.server.ServerPacket;
import net.tachyon.utils.PacketUtils;

public class PacketEncoder extends MessageToByteEncoder<ServerPacket> {

    @Override
    protected void encode(ChannelHandlerContext ctx, ServerPacket packet, ByteBuf buf) {
        PacketUtils.writePacket(buf, packet);
    }

}

File: Network/Netty/Codec/PacketEncoder.cs
using DotNetty.Buffers;
using DotNetty.Codecs;
using DotNetty.Transport.Channels;
using Tachyon.Network.Packet;
using Tachyon.Utils;

namespace Tachyon.Network.Netty.Codec;

public class PacketEncoder : MessageToByteEncoder<IServerPacket>
{
    protected override void Encode(IChannelHandlerContext context, IServerPacket message, IByteBuffer output)
    {
        BufUtil.writePacket(output, message);
    }
}